syncd/syncd_notifications.cpp

```diff
@@ -240,18 +240,71 @@ void redisPutFdbEntryToAsicView(
     g_redisClient->hset(key, strAttrId, strAttrValue);
 }
 
+void check_fdb_event_notification_data(
+        _In_ const sai_fdb_event_notification_data_t& data)
+{
+    SWSS_LOG_ENTER();
+
+    /*
+     * Any new RID value spotted in fdb notification can happen for 2 reasons:
+     *
+     * - a bug is present on the vendor SAI, all RID's are already in local or
+     *   REDIS ASIC DB but vendor SAI returned new or invalid RID
+     *
+     * - orch agent didn't query yet bridge ID/vlan ID and already
+     *   started to receive fdb notifications in which case warn message
+     *   could be ignored.
+     *
+     * If vendor SAI will return invalid RID, then this will later on lead to
+     * inconsistent DB state and possible failure on apply view after cold or
+     * warm boot.
+     *
+     * On switch init we do discover phase, and we should discover all objects
+     * so we should not get any of those messages if SAI is in consistent
+     * state.
+     */
+
+    if (!check_rid_exists(data.fdb_entry.bv_id))
+        SWSS_LOG_ERROR("bv_id RID 0x%lx is not present on local ASIC DB", data.fdb_entry.bv_id);
+
+    if (!check_rid_exists(data.fdb_entry.switch_id) || data.fdb_entry.switch_id == SAI_NULL_OBJECT_ID)
+        SWSS_LOG_ERROR("switch_id RID 0x%lx is not present on local ASIC DB", data.fdb_entry.bv_id);
+
+    for (uint32_t i = 0; i < data.attr_count; i++)
+    {
+        const sai_attribute_t& attr = data.attr[i];
+
+        auto meta = sai_metadata_get_attr_metadata(SAI_OBJECT_TYPE_FDB_ENTRY, attr.id);
+
+        if (meta == NULL)
+        {
+            SWSS_LOG_ERROR("unable to get metadata for fdb_entry attr.id = %d", attr.id);
+            continue;
+        }
+
+        // skip non oid attributes
+        if (meta->attrvaluetype != SAI_ATTR_VALUE_TYPE_OBJECT_ID)
+            continue;
+
+        if (!check_rid_exists(attr.value.oid))
+            SWSS_LOG_WARN("RID 0x%lx on %s is not present on local ASIC DB", attr.value.oid, meta->attridname);
```

Comment:
why line 268 and 271 are error level, but this one is warning level? shouldn't them be same level?

Reply:
because if received switch id in fdb notification is not existing in db already then this is a bug same for bridge vlan/id
and here is warning since user could not query yet all the bridge ports, or - thats depending on our discussion, depends how we want to handle ports, whether should we allow SAI to generate new ports on the fly